nomad/eval_endpoint.go

```diff
@@ -363,15 +363,45 @@ func (e *Eval) List(args *structs.EvalListRequest,
 			}
 
 			var evals []*structs.Evaluation
+			var itemCount int32
+			var lastToken string
+			foundToken := args.QueryOptions.LastToken == ""
 			for {
 				raw := iter.Next()
 				if raw == nil {
+					lastToken = ""
```

Comment:
Agreed. In the original design I was using `NextToken` and we _needed_ the empty string value so that we could return an empty `NextToken` when we were out of pages (we unfortunately don't have the ability to do that with the existing API). Will clean this up.